### PR TITLE
🧪 #2127 Specimen Tab updates

### PR DESCRIPTION
**Pull request by demariadaniel**
**Description of changes**
[#2127](https://app.zenhub.com/workspaces/icgc-argo-platform-dk-production-board-5e542d38415f5034e9fed89d/issues/icgc-argo/platform-ui/2127)

* Use specimen_acquisition_interval for Day Count
* Test tube icon (added in previous ticket)
* Tab has correct labels
* Specimen Files & Available Files link to File Repo w/ filter
* Pathological TNM category display
* Interval displays 'days'
* Samples, Experimental Strategies, Workflow etc. display on page

Also:
* Major updates to dummy data, built off of Data Dictionary
* Util functions for formatting Entity Data into Clinical Timeline Entity format; getting Display Names; and general data handling
* Data flow is prepped & ready for API hook up
* Move Types into root Donor folder
* Update util to .tsx

**Type of Change**

- [ ] Bug
- [ ] Styling
- [x] New Feature

**Checklist before requesting review:**

- Design (select one):
  - [x] Matches design:
    - component sizes, spacing, and styles
    - font size, weight, colour
    - spelling has been double checked
  - [ ] No design provided, screenshot of changes included in PR
  - [ ] No changes to UI
- UI Kit (select one):
  - [ ] New Components with storybook stories created
  - [ ] Modified Existing components and updates stories
  - [x] No new UIKit components or changes
- [x] Feature is minimally responsive
- [x] Manual testing of UI feature
- [x] Add copyrights to new files
- [x] Connected ticket to PR

### CODE REVIEW

File: components/pages/donor-entity/ClinicalTimeline/util.tsx [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+      );
+      return { ...sample.node, available_files };
+    });
+
+    [
+      'samples',
+      'pathological_t_category',
+      'pathological_n_category',
+      'pathological_m_category',
+    ].forEach((key) => delete data[key]);
```
Comment (ciaranschutte) [inline components/pages/donor-entity/ClinicalTimeline/util.tsx:187]: only minor suggestion would be to `[].filter` out these values and return that new array spread
not having to worry about a changing `data` object been passed around plus looks a little cleaner than blamo here's an array and we're deleting stuff. your call though :) tonne of awesome work here - loving the usage of actual types over `:any` :p

**Review by ciaranschutte — COMMENTED** [review]
one minor suggestion

**Review by ciaranschutte — APPROVED** [review]

File: components/pages/donor-entity/ClinicalTimeline/util.tsx (hunk 1 above)
Comment (demariadaniel) [reply components/pages/donor-entity/ClinicalTimeline/util.tsx:187]: Oh man I'm glad there's a better way to do this ;)
Yeah types are making a lot of sense to me now! Thanks
Very useful for keeping everything straight.

Comment (demariadaniel) [reply components/pages/donor-entity/ClinicalTimeline/util.tsx]: Updated

**Pull request merged**